cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -475,13 +476,21 @@ private void executeCondition(ApplicationSpecification appSpec, WorkflowConditio
     executeAll(iterator, appSpec, instantiator, classLoader, token);
   }
 
+  private DatasetProperties addLocalDatasetProperty(DatasetProperties properties) {
+    String dsDescription = properties.getDescription();
+    Map<String, String> dsProperties = properties.getProperties();
+    dsProperties.put(Constants.AppFabric.WORKFLOW_LOCAL_DATASET_PROPERTY, "true");
```

Comment:
isn't this unmodifiable?  we can do `addAll(properties.getProperties).add(Constants.AppFabric.WOR...)` with the builder instead.